crates/vision/src/line_detection.rs

```diff
@@ -331,110 +379,9 @@ fn is_segment_shorter_than(
-                    segment.start_edge_type = EdgeType::ImageBorder;
-                }
-                if i == number_of_segments - 1 {
-                    segment.end_edge_type = EdgeType::ImageBorder;
-                }
-                if i % 2 == 0 {
-                    segment.start_edge_type = EdgeType::Falling;
-                    segment.end_edge_type = EdgeType::Rising;
-                }
-                segments.push(segment);
-            }
-            ScanLine { position, segments }
-        }
-
-        fn create_filtered_segments(
-            number_of_scanlines: u16,
-            color: YCbCr444,
-            number_of_segments: u16,
-            segment_size: u16,
-        ) -> FilteredSegments {
-            let vertical_scan_lines = (0..number_of_scanlines)
-                .map(|position| create_scanline(color, number_of_segments, segment_size, position))
-                .collect();
-            FilteredSegments {
-                scan_grid: ScanGrid {
-                    vertical_scan_lines,
-                },
-            }
-        }
-
-        fn create_image(width: u32, height: u32) -> YCbCr422Image {
-            let width_422 = width / 2;
-            let mut buffer = vec![YCbCr422::default(); (width_422 * height) as usize];
-            for y in 0..height {
-                for x in 0..width_422 {
-                    if (y / 10) % 2 == 0 {
-                        buffer[(y * width_422 + x) as usize] = YCbCr422::new(255, 0, 0, 0);
-                    }
-                }
-            }
-            YCbCr422Image::from_ycbcr_buffer(width_422, height, buffer)
-        }
-
-        let image_size = vector![10.0, 500.0];
-
-        let image = create_image(image_size.x as u32, image_size.y as u32);
-        // let mut camera_matrix = CameraMatrix::matrix_from_parameters(
-        let camera_matrix = CameraMatrix::from_normalized_focal_and_center(
-            vector![2.0, 2.0],
-            point![1.0, 1.0],
-            image_size,
-            Isometry3 {
-                rotation: UnitQuaternion::from_euler_angles(0.0, std::f32::consts::PI / 4.0, 0.0),
-                translation: Translation::from(point![0.0, 0.0, 0.5]),
-            },
-            Isometry3::identity(),
-            Isometry3::identity(),
-        );
-        let filtered_segments =
-            create_filtered_segments(10, YCbCr444 { y: 0, cb: 0, cr: 0 }, 10, 10);
-        let check_line_segments_projection = false;
-        let maximum_projected_segment_length = 0.3;
-        let check_edge_gradient = true;
-        let gradient_alignment = -0.95;
-
-        let (line_points, _) = filter_segments_for_lines(
-            &camera_matrix,
-            &filtered_segments,
-            &image,
-            check_line_segments_projection,
-            maximum_projected_segment_length,
-            check_edge_gradient,
-            gradient_alignment,
-        );
-        assert_eq!(line_points.len(), 32);
-    }
-
```

Comment:
Why did you remove the tests? Because they failed? 😆

Reply:
Okay, please elaborate...

Reply:
Since we were previously applying the median in the same direction we were constructing segments, we were actually using a sort of moving median window.
However, since the median direction and segment direction are now orthogonal, every pixel is only inspected once (but more pixels are looked at in total) now.

Yes, we can construct new test data which would make these tests "work" again, but we already have tests without median and the medians are tested separately anyways.
Given that the two are orthogonal now, I see little value in trying to repair these tests.

Reply:
Okay, makes sence 👍🏻